File: src/algorithms/rotate-string.js
/**
 * Given two strings s and goal, returns true if and only if s can become goal
 * after some number of shifts on s.
 *
 * A shift on s consists of moving the leftmost character of s to the rightmost
 * position. For example, if s = "abcde", then it will be "bcdea" after one
 * shift.
 *
 * @format
 * @param {string} s - A string of lowercase English letters
 * @param {string} goal - A string of lowercase English letters
 * @returns {boolean} - Returns true if s can become goal after some shifts
 */

function rotateString(s, goal) {
  for (let i = 0; i < s.length; i++) {
    if (s.slice(i) + s.slice(0, i) === goal) {
      return true;
    }
  }
  return false;
}

File: src/algorithms/rotate-string.py
def rotate_string(s, goal):
    """Given two strings s and goal, return true if and only if s can become 
    goal after some number of shifts on s. A shift on s consists of moving the 
    leftmost character of s to the rightmost position. For example, if s = 
    "abcde", then it will be "bcdea" after one shift.

    Args:
        s (str): A string of lowercase English letters.
        goal (str): A string of lowercase English letters.

    Returns:
        boolean: Returns True if s can become goal after some number of shifts
        on s.
    """
    for i in range(len(s)):
        if s[i:] + s[0: i] == goal:
            return True
    return False
